src/teleport.cpp

```diff
@@ -70,19 +70,24 @@ void Teleport::addThing(Thing* thing)
 	return addThing(0, thing);
 }
 
-bool Teleport::checkInfinityLoop(Tile* destTile)
+bool Teleport::isHuge(const Tile* destTile)
 {
-	if (!destTile) {
-		return false;
-	}
-
-	if (Teleport* teleport = destTile->getTeleportItem()) {
+	Teleport* teleport = destTile->getTeleportItem();
```

Comment:
Check if `teleport` is a nullptr, since we're not guaranteed to have a teleport item on `destTile`